### ISSUE TITLE
Chatbot Workshop

### ISSUE THREAD

**Issue by AverageMarcus** [jsoxford/jsoxford.github.com#235]
(Collecting initial ideas for a Chatbot workshop/hack day in 2017. Will add more concrete info, such as dates, when it's decided.)

/cc @danielthepope 

# Possible platforms
* Telegram (personal fave as it has a lot of features)
* Slack
* Twitter
* Facebook
* Custom Pusher chat room (for getting away from having to learn a platform)

 ## Extras

* MS Cognitive Services for some machine learning responses
* Nexmo/Twilio to do call based chats

---

Depending on what people want to get out of the day (e.g. what platform people are interested in) there is several different things we could do. If there is a keen interest in Twitter bots then we could maybe ask @edent if he's interested as he has a few interesting bots (https://twitter.com/fiverfun/, https://twitter.com/RGB_Colours, https://twitter.com/Edent_Solar)

I've messaged [Martin Beeby](https://twitter.com/thebeebs) to see about the possibility about him coming to talk about the [MS bot framework](https://dev.botframework.com/). 

Feel free to add additional idea in comments and I'll collect them in this post. 😄

**Comment by edent** [comment]
Happy to talk about my bots - although they are non-interactive.  But can talk about listening to the streaming API and responding.

**Comment by AverageMarcus** [comment]
Who said chat has to be two ways? 😜

**Comment by lpmi-13** [comment]
+1 Pusher
+1 Call-based chat (visions of sugar-plumb fairies that can listen and respond somehow depending on what they hear dancing in my head...but perhaps this would be better at the language hack...)

**Comment by edent** [comment]
@lpmi-13 you could use Twilio for the voice recognition stuff. It can do basic IVR keyword detection.

**Comment by omgmog** [comment]
In Cardboctober I [played with the `speechRecognition` API a bit](https://blog.omgmog.net/post/cardboctober-09/), you should take a look at that and using it with a web-based UI.

**Comment by lpmi-13** [comment]
hmmmm, depending on the relative dates for this and the language hack, might use this a ChatOps experiment to build on. What I'm actually more interested in is getting the raw audio data and then messing around with that instead of traditional voice recognition.

So things like responding not so much based on "words", but on relative pitch distances, tone contours, stress, etc.

**Comment by AverageMarcus** [comment]
@lpmi-13 if you haven't already speak to @danielthepope about how you can do something like that with nexmo

**Comment by benfoxall** [comment]
Yo, I’m super excited about this!

I guess the actual technologies/projects can be organised in the lead up to the event (or, even left to the attendees to come up with).  Having too much defined/built in the lead up can constrain the projects, and also put more strain on the organisation.

Perhaps it would be useful to work out (and update the main ticket):

1. The name of the event (I’m a fan of _Chat Hack_, though it’s up to you)
2. A description/overview of the what people will get out of the day (like on the [summer of hacks](http://summerofhacks.io) events).
3. Some idea of dates

@danielthepope @AverageMarcus I'm going to assign this issue/event to you (so, it's kind of up to you to own & help it happen) – you okay with that?

**Comment by benfoxall** [comment]
Date wise, jsoxfordy-stuff:

* December - soon (& usually super-busy)
* January - quiet / non-drink social
* February - evening meetup
* March - render (end), maybe a meetup
* April - evening meetup
* May - weekend event (perhaps)
* June - Language hack (perhaps)
* July - no plans

…perhaps mid March?  Might be nice to have something on the lead up to [Render](http://render-conf.com/)?

**Comment by AverageMarcus** [comment]
Yeah that all sounds cool. I'll take a look at it when I'm back from the states 👍

**Comment by AverageMarcus** [comment]
I'm thinking 25th March (the weekend before Render).

Or could do April 1st (immediately following Render) as we could get some out-of-town attendees.

Thoughts?

**Comment by lpmi-13** [comment]
Would there be any interest in creating an SMS bot with some Twilio + webhooks? I'm thinking of making something that would be accessible for feature phones (read: dumb phones).

**Comment by AverageMarcus** [comment]
YES! 

I have some twilio credit remaining that I never ended up using. I also have some Nexmo credit (similar services to Twilio) to use. Happy to share it around as long as you don't go crazy 😝

**Comment by lpmi-13** [comment]
oh, and what are the chances that anyone is interested in creating a chatbot for Line?

(apparently huge in Japan/Thailand/Indonesia)

**Comment by AverageMarcus** [comment]
https://developers.line.me/messaging-api/getting-started

**Comment by lpmi-13** [comment]
ah, nvmd...Line apparently only allows "business" account chatbots, and have some ridic registration requirments (not as bad as FB, but enough to put me off).

Kik, though, could happen...
